JASP-Engine/JASP/R/binomialtest.R

```diff
@@ -15,199 +15,327 @@
+  binomialTable$addColumnInfo(name = "level",      title = "Level",      type = "string")
+  binomialTable$addColumnInfo(name = "counts",     title = "Counts",     type = "integer")
+  binomialTable$addColumnInfo(name = "total",      title = "Total",      type = "integer")
+  binomialTable$addColumnInfo(name = "proportion", title = "Proportion", type = "number", format = "sf:4")
+  binomialTable$addColumnInfo(name = "p",          title = "p",          type = "number", format = "dp:3;p:.001")
 
-	if (is.null(variables))
-		return(NULL)
+  if (options$VovkSellkeMPR) {
+    binomialTable$addColumnInfo(name = "VovkSellkeMPR", title = "VS-MPR\u002A", type = "number", format = "sf:4")
+  }
 
-  descriptivesPlotCollection <- list()
-	testValue <- options$testValue
-	confLevel <- options$descriptivesPlotsConfidenceInterval
+  if (options$confidenceInterval) {
+    binomialTable$addColumnInfo(name = "lowerCI", title = "Lower", type = "number", format = "sf:4",
+      overtitle = paste0(100 * options$confidenceIntervalInterval, "% CI for Proportion"))
+    binomialTable$addColumnInfo(name = "upperCI", title = "Upper", type = "number", format = "sf:4",
+      overtitle = paste0(100 * options$confidenceIntervalInterval, "% CI for Proportion"))
+  }
 
-  base_breaks_y <- function(x, testValue) {
-    d <- data.frame(x = -Inf, xend = -Inf, y = 0, yend = 1)
-    list(ggplot2::geom_segment(data = d, ggplot2::aes(x = x, y = y, xend = xend,
-                                            yend = yend),
-                               inherit.aes = FALSE, size = 1),
-         ggplot2::scale_y_continuous(breaks = c(0,  round(testValue,3), 1)))
+  if (!is.null(errors) && errors == "No variables") {
+    .binomEmptyMainTable(binomialTable)
+    return()
   }
 
-  pd <- ggplot2::position_dodge(0.2)
+  for (variable in options$variables) {
+    for (level in binomResults[["spec"]][["levels"]][[variable]]) {
+      row <- binomResults[["binom"]][[variable]][[level]][1:9]
+      binomialTable$addRows(row, rowNames = paste0(variable, " - ", level))
+    }
+  }
 
-	for (i in .indices(options$variables)) {
+  # Add footnote: VovkSellkeMPR
+  if (options$VovkSellkeMPR) {
+    binomialTable$addFootnote(message = .messages("footnote", "VovkSellkeMPR"), symbol = "\u002A")
+  }
 
-		var <- options$variables[[i]]
+  # Add footnote: Alternative hypothesis
+  message <- switch(binomResults[["spec"]][["hypothesisRec"]],
+                    "two.sided" = .messages("footnote", "binomNeq",     value = options$testValue),
+                    "greater"   = .messages("footnote", "binomGreater", value = options$testValue),
+                    "less"      = .messages("footnote", "binomLess",    value = options$testValue)
+  )
 
-    d <- dataset[[.v(var)]]
-    d <- d[!is.na(d)]
-		nObs <- length(d)
+  binomialTable$addFootnote(message = message, symbol = "<em>Note.</em>")
+}
 
-		descriptivesPlotList <- list()
+.binomEmptyMainTable <- function(tab) {
```

Comment:
Found by @ASarafoglou, 

Should this be a separate function? It's only used once, and the 9 is hardcoded.

Reply:
How about: https://github.com/jasp-stats/INTERNAL-jasp/issues/80